tests/sub_group_mask/sub_group_mask_reference.cpp

```diff
@@ -0,0 +1,94 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Extension Conformance Test
+//
+//  Provides tests to check sub_group_mask operator[] const and reference
+//
+*******************************************************************************/
+
+#include "sub_group_mask_common.h"
+
+#define TEST_NAME sub_group_mask_reference
+
+namespace TEST_NAMESPACE {
+
+using namespace sycl_cts;
+#ifdef SYCL_EXT_ONEAPI_SUB_GROUP_MASK
+
+struct check_result_reference {
+  bool operator()(sycl::ext::oneapi::sub_group_mask &sub_group_mask,
+                  const sycl::sub_group &) {
+    for (size_t N = 0; N < sub_group_mask.size(); N++) {
+      sycl::ext::oneapi::sub_group_mask::reference ref_to_bit =
+          sub_group_mask[sycl::id(N)];
+      // check that reference to bit have correct value
+      if (ref_to_bit != (N % 2 == 0)) return false;
+      switch (N % 5) {
+        case 0:
+          // check reference operator=(bool x)
+          // by assigning opposite value and checking corresponding bit in mask
+          ref_to_bit = (N % 2 != 0);
+          if (sub_group_mask[sycl::id(N)] != (N % 2 != 0)) return false;
+          break;
+        case 1:
+          // check reference operator=(const reference& x)
+          // by assigning reference for next bit and checking corresponding bit in mask
+          if (N == sub_group_mask.size() - 1) break;
+          ref_to_bit = sub_group_mask[sycl::id(N + 1)];
+          if (sub_group_mask[sycl::id(N)] != ((N + 1) % 2 == 0)) return false;
+          break;
+        case 2:
+          // check reference operator~()
+          if (~ref_to_bit != (N % 2 != 0)) return false;
+          break;
+        case 3:
+          // check reference operator bool()
+          if ((bool)ref_to_bit != (N % 2 == 0)) return false;
```

Comment:
You could avoid C cast
```suggestion
          if (!!ref_to_bit != (N % 2 == 0)) return false;
```

Reply:
I've applied this comment [here](https://github.com/KhronosGroup/SYCL-CTS/pull/201/commits/e67afbb4405ffd7ac21331b9d281f47407ec3ee0#diff-1b2f8ca14e7b6f512e90a7c9e0da8f23697efb62c2c3c4eee61a29810e5d6f82L46-L47)